app/src/main/java/br/com/drss/pokedex/home/repository/PokemonRepository.kt

```diff
@@ -0,0 +1,76 @@
+package br.com.drss.pokedex.home.repository
+
+import br.com.drss.pokedex.home.repository.database.PokemonSummaryDao
+import br.com.drss.pokedex.home.repository.domain.entities.PokemonSummary
+import br.com.drss.pokedex.home.repository.domain.entities.PokemonType
+import br.com.drss.pokedex.home.repository.network.PokeApi
+import br.com.drss.pokedex.home.repository.network.dto.PokemonDto
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+
+interface PokemonRepository {
+
+    fun getPokemonSummaryList(typeFilters: List<PokemonType> = listOf()): Flow<List<PokemonSummary>>
+    suspend fun getPokemonSummary(name: String): PokemonSummary
+
+}
+
+class PokemonRepositoryImpl(
```

Comment:
Personally I dont like "impl" in names. Basically means that you had an abstraction with only one implementation possible.